iceoryx2-cli/iox2/src/commands.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use colored::*;
+use std::env;
+use std::fs;
+use std::io;
+use std::path::PathBuf;
+use std::process::{Command, Stdio};
+use thiserror::Error;
+
+#[derive(Clone, Debug)]
+struct CommandInfo {
+    name: String,
+    path: PathBuf,
+    is_development: bool,
+}
+
+#[derive(Error, Debug)]
+pub enum ExecutionError {
+    #[error("Command not found: {0}")]
+    NotFound(String),
+    #[error("Execution failed: {0}")]
+    Failed(String),
+}
+
+pub fn list() {
+    println!("{}", "Installed Commands:".bright_green().bold());
+    let mut installed_commands = find();
+    installed_commands.sort_by_key(|command| command.name.clone());
+    installed_commands
+        .iter()
+        .map(|command| {
+            format!(
+                "  {} {}",
+                command.name.bold(),
+                if command.is_development {
+                    "(dev) ".italic()
+                } else {
+                    "".italic()
+                },
+            )
+        })
+        .for_each(|formatted_command| println!("{}", formatted_command));
+}
+
+fn get_development_binaries_dir() -> Result<std::path::PathBuf, io::Error> {
```

Comment:
Doesn't this return a directory to the dev binaries? If so maybe the name `get_dev_binary_path` is more suitable.

Reply:
@elfenpiff I don't see the big difference here... besides using `dir` vs. `path`.

Reply:
@orecham you are right!